src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -1549,6 +1549,8 @@ impl Step for MirOpt {
             })
         };
 
+        run(self.target);
+
         if builder.config.cmd.bless() {
```

Comment:
```suggestion
        // Run more targets with `--bless`. But we always run the host target first, since some
        // tests use very specific `only` clauses that are not covered by the target set below.
        if builder.config.cmd.bless() {
```